lib/client.js

```diff
@@ -572,11 +572,14 @@ class Client {
   search({ resourceType, compartment, searchParams, headers } = {}) {
     if (compartment && resourceType) {
       return this.compartmentSearch({ resourceType, compartment, searchParams, headers });
-    } else if (resourceType && searchParams) {
-      return this.resourceSearch({ resourceType, searchParams, headers });
+    } else if (resourceType) {
+      const params = searchParams || {};
+      return this.resourceSearch({ resourceType, params, headers });
     } else if (searchParams) {
       return this.systemSearch({ searchParams, headers });
     }
+
+    throw new Error('search requires either searchParams or a resourceType');
```

Comment:
A test should be added for this error, and the jsdoc for this function should be updated with a `@throws` description.

I don't think the error text is quite right. I think the requirements are:
```
(compartment && resourceType) || searchParams
```
Looking at this made me realize that we don't support a compartment search without a `resourceType`, but I'll make a separate issue for that.